platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java

```diff
@@ -83,12 +89,10 @@ public void initialize() {
     serviceState = State.INITIALIZING;
 
     LOGGER.info("Starting the service with this config {}", appConfig);
-    doInit();
-
     Config metricsConfig = appConfig.hasPath(METRICS_CONFIG_KEY) ?
         appConfig.getConfig(METRICS_CONFIG_KEY) : ConfigFactory.empty();
-    PlatformMetricsRegistry.initMetricsRegistry(getServiceName(), metricsConfig);
-
+    PlatformMetricsRegistry.initMetricsRegistry(this.appConfig.getString(SERVICE_NAME_CONFIG), metricsConfig);
```

Comment:
I think you should pass `getServiceName()`'s output only. Otherwise, you're not letting the services to chose their own name.

Reply:
done